dipy/denoise/noise_estimate.py

```diff
@@ -88,20 +187,28 @@ def piesno(data, N=1, alpha=0.01, l=100, itermax=100, eps=1e-5, return_mask=Fals
 
     if N in opt_quantile:
         q = opt_quantile[N]
+
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give
+    # up: 
+    if m == 0:
```

Comment:
Sorry, I am not sure I understand.

These lines are identical in this PR, and in the PR you did against this
branch, here:

https://github.com/arokem/dipy/pull/13

Do you have a further suggestion for change? Should we change this line to
the following?

`if m==0 and np.all(data==0)`



On Tue, Feb 24, 2015 at 1:47 PM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/denoise/noise_estimate.py
> <https://github.com/nipy/dipy/pull/576#discussion_r25296019>:
>
> >      # Initial estimation of sigma
> >      denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
> >      m = np.percentile(data, q * 100) / denom
> > +
> > +    # if the percentile is 0, then more than half of the slice is zero and give
> > +    # up:
> > +    if m == 0:
>
> Only thing missing if if we really want to have this check, or instead
> have a sanity check for when the whole slice is zero (and not only the
> quantile).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/576/files#r25296019>.
>